js/apps/admin-ui/src/groups/Members.tsx

```diff
@@ -87,14 +88,29 @@ export const Members = () => {
   const getMembership = async (id: string) =>
     await adminClient.users.listGroups({ id: id! });
 
-  const getSubGroups = (groups: GroupRepresentation[]) => {
-    let subGroups: GroupRepresentation[] = [];
-    for (const group of groups!) {
-      subGroups.push(group);
-      const subs = getSubGroups(group.subGroups!);
-      subGroups = subGroups.concat(subs);
+  // this queries the subgroups using the new search paradigm but doesn't
+  // account for pagination and therefore isn't going to scale well
+  const getSubGroups = async (groupId: string, count: number = 0) => {
+    let nestedGroups: GroupRepresentation[] = [];
+    if (!count) {
+      return nestedGroups;
+    } else {
+      const params: Record<string, string> = {
+        first: "0",
+        max: count.toString(),
+      };
+      const subGroups: GroupRepresentation[] = await fetchAdminUI<
+        GroupRepresentation[]
+      >(`groups/${groupId}/children`, { ...params });
+      nestedGroups = nestedGroups.concat(subGroups);
+      for (const g of subGroups) {
+        if (g.id) {
+          const additionalGroups = await getSubGroups(g.id, g.subGroupCount);
```

Comment:
This should be done concurrently using `Promise.all()`, awaiting like this will mean that each request in this loop blocks the other until the previous request has finished.